go-controller/pkg/factory/factory.go

```diff
@@ -479,6 +483,97 @@ func NewWatchFactory(c kubernetes.Interface, stopChan chan struct{}) (*WatchFact
 	return wf, nil
 }
 
+// NewNodeWatchFactory creates only the informers we need for node-level operations,
+// saving bandwidth and memory
+// It only configures informers for:
+// - pods, and only the pods on this node
+// - services
+// - endpoints
+// - nodes
+func NewNodeWatchFactory(nodeName string, c kubernetes.Interface, stopChan chan struct{}) (*WatchFactory, error) {
+	// resync time is 12 hours, none of the resources being watched in ovn-kubernetes have
+	// any race condition where a resync may be required e.g. cni executable on node watching for
+	// events on pods and assuming that an 'ADD' event will contain the annotations put in by
+	// ovnkube master (currently, it is just a 'get' loop)
+	// the downside of making it tight (like 10 minutes) is needless spinning on all resources
+	wf := &WatchFactory{
+		iFactory:  informerfactory.NewSharedInformerFactory(c, resyncInterval),
+		informers: make(map[reflect.Type]*informer),
+	}
+	var err error
+
+	// Create informers with node-specific filters
+
+	// Pod informer: we only want pods scheduled to this node
+	wf.informers[podType], err = newInformer(podType, v1coreinformers.NewFilteredPodInformer(
+		c, "", resyncInterval,
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		func(opts *metav1.ListOptions) {
+			opts.FieldSelector = fields.OneTermEqualSelector("spec.nodeName", nodeName).String()
+		}))
+	if err != nil {
+		return nil, err
+	}
+
+	// services, endpoints: skip headless and other uninteresting services
+	// taken from k8s cmd/kube-proxy/app/server.go
+	noProxyName, err := labels.NewRequirement("service.kubernetes.io/service-proxy-name", selection.DoesNotExist, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	noHeadlessEndpoints, err := labels.NewRequirement(v1.IsHeadlessService, selection.DoesNotExist, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	labelSelector := labels.NewSelector()
+	labelSelector = labelSelector.Add(*noProxyName, *noHeadlessEndpoints)
+
+	// services: skip headless services
+	wf.informers[serviceType], err = newInformer(serviceType, v1coreinformers.NewFilteredServiceInformer(
+		c, "", resyncInterval,
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		func(opts *metav1.ListOptions) {
+			opts.LabelSelector = labelSelector.String()
+		}))
+	if err != nil {
+		return nil, err
+	}
+	wf.informers[endpointsType], err = newInformer(endpointsType, v1coreinformers.NewFilteredEndpointsInformer(
+		c, "", resyncInterval,
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		func(opts *metav1.ListOptions) {
+			opts.LabelSelector = labelSelector.String()
+		}))
+	if err != nil {
+		return nil, err
+	}
+
+	wf.informers[nodeType], err = newQueuedInformer(nodeType, wf.iFactory.Core().V1().Nodes().Informer(), stopChan)
+	if err != nil {
+		return nil, err
+	}
+
+	wf.iFactory.Start(stopChan)
```

Comment:
stuff after here should maybe be shared rather than copied+pasted?

Reply:
done